### PR TITLE
Added unique title validation for News entities

### PR DESCRIPTION
**Pull request by darkravchuk**
### Summary of Issue
The main issue involves adding validation to ensure that News entities have unique titles within the system. The current implementation lacks a check, which allows duplicate news entries with the same title, leading to potential conflicts and confusion when managing content.

### Summary of Change
- Implemented a validation check that prevents duplicate titles when creating or updating a News entity.
- Modified the form to validate title uniqueness before submission.
- Adjusted the error handling logic to notify the user if a duplicate title is detected.

### CODE REVIEW

File: src/features/AdminPage/NewsPage/NewsModal/NewsModal.component.tsx [hunk 1]
```diff
@@ -94,6 +94,15 @@ const NewsModal: React.FC<{
...
 
+    const checkUniqueTitle = async (title: string): Promise<boolean> => {
+        const newsList = newsStore.NewsArray;
+        if(newsItem) {
+            const filteredNewsList = newsList.filter((news: News) => news.id !== newsItem?.id);
+            return filteredNewsList.every((news: News) => news.title !== title);
+        }
+        return newsList.every((news: News) => news.title !== title);
+    }
+
```
Comment (sashapanasiuk5) [inline src/features/AdminPage/NewsPage/NewsModal/NewsModal.component.tsx:105]: We already have generic implemetation for this type of checks. It is in 'app/common/utils/uniquenessValidator'
Maybe you find it useful)

File: yarn.lock [hunk 2]
```diff
@@ -6542,6 +6542,11 @@ fs.realpath@^1.0.0:
   resolved "https://registry.npmjs.org/fs.realpath/-/fs.realpath-1.0.0.tgz"
   integrity sha512-OO0pH2lK6a0hZnAdau5ItzHPI6pUlvI7jMVnxUQRtw4owF2wk8lOSabtGDCTP4Ggrg2MbGnWO9X8K1t4+fGMDw==
 
+fsevents@^2.3.2, fsevents@~2.3.2:
+  version "2.3.3"
+  resolved "https://registry.npmjs.org/fsevents/-/fsevents-2.3.3.tgz"
+  integrity sha512-5xoDfX+fL7faATnagmWPpbFtwh/R77WmMMqqHGS65C3vvB0YHrgF+B1YmZ3441tMj5n63k0212XNoJwzlhffQw==
+
```
Comment (sashapanasiuk5) [inline yarn.lock:6549]: Do we really need this dependency?

File: src/features/AdminPage/NewsPage/NewsModal/NewsModal.component.tsx (hunk 1 above)
Comment (darkravchuk) [reply src/features/AdminPage/NewsPage/NewsModal/NewsModal.component.tsx:105]: You're right, I didn't notice the existing generic implementation. I'll update it to use 'uniquenessValidator'. Thanks for pointing it out!

File: yarn.lock (hunk 2 above)
Comment (darkravchuk) [reply yarn.lock:6549]: fsevents was added automatically as it's required by some packages like jest and react-scripts for file watching on macOS.

**Review by sashapanasiuk5 — APPROVED** [review]

**Review by Michael-Kolpakov — APPROVED** [review]

**Review by ulyasaur — APPROVED** [review]

**Pull request merged**